components/server/src/authorization/spicedb.ts

```diff
@@ -36,49 +66,106 @@ export function spiceDBConfigFromEnv(): SpiceDBClientConfig | undefined {
 
 export class SpiceDBClientProvider {
     private client: Client | undefined;
+    private previousClientOptionsString: string = DEFAULT_FEATURE_FLAG_VALUE;
+    private clientOptions: grpc.ClientOptions;
 
     constructor(
         private readonly clientConfig: SpiceDBClientConfig,
         private readonly interceptors: grpc.Interceptor[] = [],
-    ) {}
+    ) {
+        this.clientOptions = DefaultClientOptions;
+        this.reconcileClientOptions()
+            .then(() => {})
+            .catch((e) => {
+                log.error("[spicedb] Failed to reconcile client options", e);
+            });
+    }
 
-    getClient(): SpiceDBClient {
-        if (!this.client) {
-            this.client = v1.NewClient(
+    private async reconcileClientOptions() {
+        const doReconcileClientOptions = async () => {
+            try {
+                const customClientOptions = await getExperimentsClientForBackend().getValueAsync(
+                    "spicedb_client_options",
+                    DEFAULT_FEATURE_FLAG_VALUE,
+                    {},
+                );
+                if (customClientOptions === this.previousClientOptionsString) {
+                    return;
+                }
+                let clientOptions = DefaultClientOptions;
+                if (customClientOptions && customClientOptions != DEFAULT_FEATURE_FLAG_VALUE) {
+                    clientOptions = JSON.parse(customClientOptions);
+                }
+                if (this.client != null) {
+                    const newClient = this.createClient(clientOptions);
+                    const oldClient = this.client;
+                    this.client = newClient;
+
+                    log.info("[spicedb] Client options changes", {
+                        clientOptions: new TrustedValue(clientOptions),
+                    });
+
+                    // close client after 10s to make sure most pending requests on the previous client are finished.
+                    setTimeout(() => {
+                        this.closeClient(oldClient);
+                    }, 10 * 1000);
```

Comment:
Let's use a longer timeout here, e.g. maybe 1-2 minutes?
Reasoning: our client timeouts are way shorter (8s I believe?). And we should not constraint ourselves here too much, so we have some leeway we can use to fine-tune the config itself. This is the catch-all here anyway, only here for the worst case.